### PR TITLE
Fix application opened

### PR DESCRIPTION
**Pull request by tonyxiao**
**What does this PR do?**
`Application Opened` is now sent when application moves from background to forground. 
`Application Opened` now includes properties listed in our mobile spec (`from_background`, `referring_application`,`url`) when application available.  For example, when launching app from a URL. 

**What are the relevant tickets?**
https://segment.atlassian.net/browse/EPD-1153

**Questions:**
- Does the docs need an update?
Application Backgrounded: https://github.com/segmentio/site-docs/pull/2158/
Application Opened was not updated in Mobile Spec as it is not consistent with Android functionality

@segmentio/gateway
cc @teresanesteby

### CODE REVIEW

**Comment by f2prateek** [comment]
1. Can we start adding tests for these?
2. The original behaviour was put in place so that even if you don't initialize segment in `didFinishLaunch` phase, we could track an 'App Opened' event. Even though that's what's recommended, people don't always follow it. I agree that this is the right behaviour, so we should document this for `trackApplicationLifecycleEvents` setter and in site-docs.

**Comment by tonyxiao** [comment]
@f2prateek do you have a good pattern for testing these hooks? And yes agreed we should update the docs @TeresaNesteby

**Comment by TeresaNesteby** [comment]
@f2prateek @tonyxiao I am working on the ios lib docs update and will link here when I have a PR.

**Comment by f2prateek** [comment]
@tonyxiao with the middleware change, I think we should be able to have a "test" middleware that we can verify is invoked with this event.

**Comment by tonyxiao** [comment]
True. Will work on that with @TeresaNesteby on thursday.

File: Analytics/Classes/SEGAnalytics.m [hunk 1]
```diff
@@ -99,13 +97,24 @@ - (void)dealloc
...
+        [self _applicationWillEnterForeground];
     }
+}
 
-    // Previously SEGBuildKey was stored an integer. This was incorrect because the CFBundleVersion
+- (void)_applicationDidFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    if (!self.configuration.trackApplicationLifecycleEvents) {
+        return;
+    }
+        // Previously SEGBuildKey was stored an integer. This was incorrect because the CFBundleVersion
```
Comment (f2prateek) [inline Analytics/Classes/SEGAnalytics.m]: I think the formatting is off here. I just run https://github.com/square/spacecommander to clean it up.

Comment (tonyxiao) [inline Analytics/Classes/SEGAnalytics.m]: @TeresaNesteby wanna take a stab at this? :)

**Comment by tonyxiao** [comment]
@f2prateek how come closed?

**Comment by f2prateek** [comment]
@tonyxiao Sorry, I accidentally deleted the branch as I was pruning old ones. Restored it but forgot to reopen.

File: Analytics/Classes/SEGAnalytics.m (hunk 1 above)
Comment (TeresaNesteby) [inline Analytics/Classes/SEGAnalytics.m]: @f2prateek @tonyxiao Just pushed up updates to indentation. I used Xcode's default to reindent.  Let me know if it doesn't match up with what you are seeing with spacecommander.

File: Analytics/Classes/SEGAnalytics.m [hunk 2]
```diff
@@ -164,10 +164,10 @@ - (void)_applicationWillEnterForeground {
     NSString *currentVersion = [[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"];
     NSString *currentBuild = [[NSBundle mainBundle] infoDictionary][@"CFBundleVersion"];
     [self track:@"Application Opened" properties:@{
-        @"from_background": @YES,
-        @"version" : currentVersion,
-        @"build" : currentBuild,
-    }];
+                                                   @"from_background": @YES,
```
Comment (tonyxiao) [inline Analytics/Classes/SEGAnalytics.m]: @TeresaNesteby did you re-format this by hand or use a tool? New indentation seems off actually

Comment (TeresaNesteby) [inline Analytics/Classes/SEGAnalytics.m]: @tonyxiao I used Xcode - structure - re-indent. I'll try using spacecommander - that prateek mentioned.

File: Analytics/Classes/SEGAnalytics.m (hunk 1 above)
Comment (f2prateek) [inline Analytics/Classes/SEGAnalytics.m]: You can use this via xcode https://github.com/travisjeffery/ClangFormat-Xcode. Iirc it has the same output as space commander.

**Comment by tonyxiao** [comment]
fyi @TeresaNesteby @f2prateek

**Comment by f2prateek** [comment]
LGTM!

**Pull request merged**